### PR TITLE
Restore mistakenly removed public APIs

### PR DESCRIPTION
**Pull request by amcasey**
As per https://github.com/dotnet/roslyn/issues/7989, we never marked our 1.1 APIs shipped, so I didn't realize I couldn't remove `CSharpObjectFormatter.Instance` and `VisualBasicObjectFormatter.Instance`.

### CODE REVIEW

**Comment by amcasey** [comment]
@mattwar This makes a "Common" type public. :(  Suggestions welcome.

**Comment by amcasey** [comment]
@tmat FYI

**Comment by amcasey** [comment]
FYI @dotnet/roslyn-interactive

**Comment by tmat** [comment]
Did we ship public CommonObjectFormatter? Oh, well.

**Comment by amcasey** [comment]
TODO: also restore
```
        public string FormatObject(object obj) => FormatObject(obj, new PrintOptions());
```

**Comment by amcasey** [comment]
@tmat, no it hasn't shipped yet.  It's a new layer.

**Comment by amcasey** [comment]
test prtest/lin/dbg/unit32 please

**Comment by amcasey** [comment]
test prtest/mac/dbg/unit32 please

**Comment by mattwar** [comment]
@amcasey  you can keep the common stuff internal and make a special CSharpObjectFormatterImpl class that extends it.  Then make CSharpObjectFormatter class a wrapper over CSharpObjectFormatterImp that defers calls to the inner instance.

**Comment by amcasey** [comment]
@mattwar Working on it now.

**Comment by amcasey** [comment]
Updated PR

File: src/Scripting/CSharp/Hosting/ObjectFormatter/CSharpObjectFormatter.cs [hunk 1]
```diff
@@ -1,30 +1,22 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Reflection;
+using System;
 using Microsoft.CodeAnalysis.Scripting.Hosting;
 
 namespace Microsoft.CodeAnalysis.CSharp.Scripting.Hosting
 {
-    internal class CSharpObjectFormatter : CommonObjectFormatter
+    public class CSharpObjectFormatter : ObjectFormatter
```
Comment (mattwar) [inline src/Scripting/CSharp/Hosting/ObjectFormatter/CSharpObjectFormatter.cs]: Sealed now?

**Comment by amcasey** [comment]
@mattwar Done

**Pull request merged**